dask/dataframe/groupby.py

```diff
@@ -797,17 +858,29 @@ def _aca_agg(self, token, func, aggfunc=None, split_every=None,
             aggfunc = func
 
         meta = func(self._meta)
-        columns = meta.name if is_series_like(meta) else meta.columns
+        if self._slice is None:
+            # pandas shows different behaviour for observed=True if a slice
```

Comment:
Do you have an example? This sounds somewhat like a bug in pandas.

Reply:
Hi Tom. Thank you for taking your time to review this PR.
A MWE is the following (with pandas version 0.23.4):

```
import pandas as pd
df = pd.DataFrame({
        'c1': pd.Categorical(['v11', 'v12', 'v11'], ['v11', 'v12', 'v13']),
        'c2': pd.Categorical(['v21', 'v22', 'v23']),
        'c3': [1, 4, 8]})

df.groupby(['c1', 'c2'], observed=True).sum()
```

This returns, as expected:

c1|c2|c3
--|--|--
v11|v21|1
v11|v23|8
v12|v22|4

Whereas

```
df.groupby(['c1', 'c2'], observed=True)[['c3']].sum()
```

returns:

c1|c2|c3
--|--|--
v11|v21|1.0
v11|v22|
v11|v23|8.0
v12|v21|
v12|v22|4.0
v12|v23|

I haven't submitted the issue to pandas because in pandas 0.24.0rc1 the behavior is consistent.

Reply:
Could you consolidate this into a method on _Groupby

```
columns = self._get_columns(meta)
```

and include the note about it being possible to remove the slice condition for pandas>=0.24